search/search_indexes.py

```diff
@@ -14,11 +14,82 @@
 
 from __future__ import unicode_literals
 
-from lxml import etree
+from collections import defaultdict
+from datetime import datetime, timedelta
+import logging
+
+from django.conf import settings
 from haystack import indexes
+from lxml import etree
+
+from learningresources.models import Course, LearningResource
+
+INDEX_CACHE = {
+    "course": {},
+    "term": {},
+    "born": datetime.now(),
+}
+MAX_INDEX_AGE = timedelta(minutes=1)
+
+log = logging.getLogger(__name__)
+
+
+def check_cache_age():
+    """
+    Keep the cache from getting stale.
 
-from learningresources.models import LearningResource
-from taxonomy.models import Vocabulary
+    Must not cache during tests or development, or it will
+    cause confusion. Must manually enable caching in production.
+    """
+    # pylint: disable=global-statement
+    global INDEX_CACHE
+    if (datetime.now() - INDEX_CACHE["born"] > MAX_INDEX_AGE) \
+            or settings.ALLOW_CACHING is False:
+        INDEX_CACHE = {
+            "course": {},
+            "term": {},
+            "born": datetime.now(),
+        }
+
+
+def get_course_metadata(course_id):
```

Comment:
The problem with memoize is that there's no concept of expiration. I do think Django's caching would be an upgrade, since they [added the ability](https://docs.djangoproject.com/en/1.8/topics/cache/#cache-arguments) to set an expiration timeout in Django 1.7. I'll open an issue for it.